tests/manager/api/controller/test_scopedsession.py

```diff
@@ -1,68 +1,62 @@
+from collections.abc import Generator
 from contextlib import contextmanager
+from unittest.mock import MagicMock
 
 import flask
+import pytest
 from sqlalchemy.orm import Session
+from typing_extensions import Self
 
-from palace.manager import api
-from palace.manager.core.opds_import import OPDSAPI
+from palace.manager.api.app import app, initialize_database
 from palace.manager.sqlalchemy.flask_sqlalchemy_session import current_session
-from palace.manager.sqlalchemy.model.collection import Collection
 from palace.manager.sqlalchemy.model.datasource import DataSource
 from palace.manager.sqlalchemy.model.identifier import Identifier
-from palace.manager.sqlalchemy.model.library import Library
-from palace.manager.sqlalchemy.util import create
-from tests.fixtures.api_controller import (
-    ControllerFixture,
-    ControllerFixtureSetupOverrides,
-)
-
-
-class ScopedHolder:
-    """A scoped holder used to store some state in the test. This is necessary because
-    we want to do some unusual things with scoped sessions, and don't necessary have access
-    to a database transaction fixture in all of the various methods that will be called.
-    """
+from tests.fixtures.database import DatabaseFixture
+from tests.fixtures.services import ServicesFixture
+from tests.mocks.circulation import MockCirculationManager
+
+
+class ScopedSessionFixture:
+    def __init__(
+        self, db_fixture: DatabaseFixture, services: ServicesFixture, session: Session
```

Comment:
This tests was largely re-written. It had to jump through a lot of hoops before, since the assumption was that it ran inside of a transaction that could be rolled back at the end. That made the functionality hard to test. 

Now that we have a fixture `function_database` that provides a temporary database for the test function, we can re-write this to do the test more directly. This fixture is needed for the migration tests anyway, as they can't be written to work in a transaction, so it makes sense to use it here as well.